modules/programs/zsh.nix

```diff
@@ -34,6 +34,21 @@ let
 
   historyModule = types.submodule ({ config, ... }: {
     options = {
+      append = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          If set, zsh sessions will append their history list to the history
+          file, rather than replace it. Thus, multiple parallel zsh sessions
+          will all have the new entries from their history lists added to the
+          history file, in the order that they exit.
+
+          This file will still be periodically re-written to trim it when the
+          number of lines grows 20% beyond the value specified by
+          `programs.zsh.history.save`.
```

Comment:
do we have `programs.zsh.history.save` in HM ?

Reply:
Yep. It's an option that is defined a few lines down:

https://github.com/nix-community/home-manager/pull/5457/files#diff-de06c2704c11a6d97ba4e8e063647a69aba101cb056fd55fdb9b287481c74a34R58